### PR TITLE
Rewrite UIKit tests

### PR DESCRIPTION
**Pull request by frootloops**
#### What's in this pull request?

I rewrite the test file but I'm not sure that it works because when I run `../llvm/utils/lit/lit.py -sv  ../build/Ninja-DebugAssert/swift-macosx-x86_64/test-iphonesimulator-i386/1_stdlib/UIKit.swift` I got 
`
/Users/arsen/dev/apple/swift/test/1_stdlib/UIKit.swift:6:8: error: no such module 'StdlibUnittest'
import StdlibUnittest
`

So I hope you could give me an advice about how can I make sure that it works. Thanks 🤖

### CODE REVIEW

File: test/1_stdlib/UIKit.swift [hunk 1]
```diff
@@ -1,77 +1,86 @@
-// RUN: %target-run-simple-swift | FileCheck %s
+// RUN: %target-run-stdlib-swift
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: Why this change in the run line?

**Comment by gribozavr** [comment]
@swift-ci Please test

File: test/1_stdlib/UIKit.swift [hunk 2]
```diff
@@ -1,77 +1,86 @@
...
+  var s = "\(o.isPortrait) \(UIDeviceOrientationIsPortrait(o)), "
+  s += "\(o.isLandscape) \(UIDeviceOrientationIsLandscape(o)), "
+  s += "\(o.isFlat), "
+  s += "\(o.isValidInterfaceOrientation) \(UIDeviceOrientationIsValidInterfaceOrientation(o))"
+  return s
+}
+
+UIKitTests.test("Device orientations") {
+  var s = printOrientation(.unknown)
+  expectEqual("false false, false false, false, false false", s)
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: Why not `expectEqual(..., printOrientation(...))` and avoid the mutable variable?

File: test/1_stdlib/UIKit.swift [hunk 3]
```diff
@@ -1,77 +1,86 @@
...
+
+  s = printOrientation(.landscapeRight)
+  expectEqual("false false, true true", s)
+}
+
+UIKitTests.test("UIEdgeInsets") {
+  let inset1 = UIEdgeInsets(top: 1.0, left: 2.0, bottom: 3.0, right: 4.0)
+  let inset2 = UIEdgeInsets(top: 1.0, left: 2.0, bottom: 3.1, right: 4.0)
+  expectEqual(inset1, inset1)
+  expectNotEqual(inset1, inset2)
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]:
```
let instances = [ UIEdgeInsets(..., 3.0, ...), UIEdgeInsets(..., 3.1, ...), UIEdgeInsets.zero ]
checkEquatable(instances, oracle: { $0 == $1 })
```

This checks more combinations.

File: test/1_stdlib/UIKit.swift [hunk 4]
```diff
@@ -1,77 +1,86 @@
...
+UIKitTests.test("UIOffset") {
+  var offset1 = UIOffset(horizontal: 1.0, vertical: 2.0)
+  var offset2 = UIOffset(horizontal: 1.0, vertical: 3.0)
+  expectEqual(offset1, offset1)
+  expectNotEqual(offset1, offset1)
+  expectEqual(offset1, offset2)
+
+  let offset0 = UIOffset(horizontal: 0.0, vertical: 0.0)
+  let offsetDot0 = UIOffset.zero
+  expectEqual(offset0, offsetDot0)
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: `checkEquatable` again.

File: test/1_stdlib/UIKit.swift [hunk 5]
```diff
@@ -1,77 +1,86 @@
...
+import StdlibUnittest
 
-func printOrientation(o: UIDeviceOrientation) {
-  print("\(o.isPortrait) \(UIDeviceOrientationIsPortrait(o)), ", terminator: "")
-  print("\(o.isLandscape) \(UIDeviceOrientationIsLandscape(o)), ", terminator: "")
-  print("\(o.isFlat), ", terminator: "")
-  print("\(o.isValidInterfaceOrientation) \(UIDeviceOrientationIsValidInterfaceOrientation(o))")
+let UIKitTests = TestSuite("UIKit")
+
+private func printOrientation(o: UIDeviceOrientation) {
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: `-> String`

File: test/1_stdlib/UIKit.swift [hunk 6]
```diff
@@ -1,77 +1,86 @@
...
+  expectEqual("false false, true true, false, true true", s)
+
+  s = printOrientation(.faceUp)
+  expectEqual("false false, false false, true, false false", s)
+
+  s = printOrientation(.faceDown)
+  expectEqual("false false, false false, true, false false", s)
+}
+
+private func printOrientation(o: UIInterfaceOrientation) {
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: `-> String`

**Comment by gribozavr** [comment]
> I got `/Users/arsen/dev/apple/swift/test/1_stdlib/UIKit.swift:6:8: error: no such module 'StdlibUnittest' import StdlibUnittest `

Did you run `build-script` with `-i` to build modules for iOS?

**Comment by frootloops** [comment]
@gribozavr could you have a look at the PR

File: test/1_stdlib/UIKit.swift [hunk 7]
```diff
@@ -1,77 +1,77 @@
...
-
-
-func printOrientation(o: UIInterfaceOrientation) {
-  print("\(o.isPortrait) \(UIInterfaceOrientationIsPortrait(o)), ", terminator: "")
-  print("\(o.isLandscape) \(UIInterfaceOrientationIsLandscape(o))")
+UIKitTests.test("UIOffset") {
+  let offsets = [ UIOffset(horizontal: 1.0, vertical: 2.0),
+    UIOffset(horizontal: 1.0, vertical: 3.0), UIOffset.zero ]
+  checkEquatable(offsets, oracle: { $0 == $1 })
+  expectEqual(UIOffset.zero, UIOffset.zero)
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: You shouldn't need that last `expectEqual()` check here (and in the test above).  `checkEquatable` should be doing it.

File: test/1_stdlib/UIKit.swift [hunk 8]
```diff
@@ -1,77 +1,77 @@
...
+  expectEqual("false false, true true",
+    printInterface(.landscapeLeft))
+
+  expectEqual("false false, true true",
+    printInterface(.landscapeRight))
+}
+
+UIKitTests.test("UIEdgeInsets") {
+  let insets = [ UIEdgeInsets(top: 1.0, left: 2.0, bottom: 3.0, right: 4.0),
+    UIEdgeInsets(top: 1.0, left: 2.0, bottom: 3.1, right: 4.0), UIEdgeInsets.zero ]
```
Comment (gribozavr) [inline test/1_stdlib/UIKit.swift]: It is more conventional to put one array item per line:

```swift
let insets = [
  UIEdgeInsets(top: 1.0, left: 2.0, bottom: 3.0, right: 4.0),
  UIEdgeInsets(top: 1.0, left: 2.0, bottom: 3.1, right: 4.0),
  UIEdgeInsets.zero
]
```

**Comment by gribozavr** [comment]
@frootloops LGTM modulo two minor comments.  Triggering the CI anyway to get more feedback.

@swift-ci Please test

**Comment by frootloops** [comment]
@gribozavr done

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
The failure looks unrelated.  Thank you!
